src/interfaceCallForwarder.ts

```diff
@@ -0,0 +1,395 @@
+      id,
+      ast,
+      struct_names,
+      true,
+    );
+    functions.push(
+      new FunctionDefinition(
+        id,
+        '',
+        sourceUnit.id,
+        FunctionKind.Free,
+        element.name,
+        false,
+        FunctionVisibility.Internal,
+        FunctionStateMutability.NonPayable,
+        false, // is constructor
+        new ParameterList(ast.reserveId(), '', inputParameters), // parameters
+        new ParameterList(ast.reserveId(), '', returnParameters), // return parameters
+        [], // modifiers
+        undefined, // override specifier
+        new Block(ast.reserveId(), '', []), // body
+        ['warp-cairo', ...element.stub].join('\n'), // documentation
+      ),
+    );
+  });
+  functions.forEach((f: FunctionDefinition) => {
+    sourceUnit.appendChild(f);
+    ast.setContextRecursive(f);
+  });
+}
+
+class CairoContractSolWriter extends ASTNodeWriter {
+  writeInner(node: CairoContract, writer: ASTWriter): SrcDesc {
+    const result: SrcDesc = [];
+
+    const solContract = new ContractDefinition(
+      node.id,
+      node.src,
+      node.name,
+      node.scope,
+      node.kind,
+      node.abstract,
+      node.fullyImplemented,
+      node.linearizedBaseContracts,
+      node.usedErrors,
+      node.documentation,
+      node.children,
+      node.nameLocation,
+      node.raw,
+    );
+    result.push(writer.write(solContract).slice(0, -1));
+    result.push('\n    // Write your logic here\n');
+    result.push('}\n');
+    return result;
+  }
+}
+
+class CornerTypeNameWriter extends ASTNodeWriter {
+  writeInner(node: CornerTypeName, _writer: ASTWriter): SrcDesc {
+    const result: SrcDesc = [];
+    result.push(node.typeString);
+    return result;
+  }
+}
+
+class CornerStructTypeNameWriter extends ASTNodeWriter {
+  writeInner(node: CornerTypeName, _writer: ASTWriter): SrcDesc {
+    const result: SrcDesc = [];
+    result.push(node.typeString);
+    return result;
+  }
+}
+
+class CornerTypeName extends TypeName {
+  constructor(id: number, src: string, typeString: string) {
+    super(id, src, typeString);
+  }
+}
+
+class CornerStructTypeName extends TypeName {
+  constructor(id: number, src: string, typeString: string) {
+    super(id, src, typeString);
+  }
+}
+
+function getSolTypeName(cairoTypeString: string, ast: AST, struct_names: string[]): TypeName {
+  if (cairoTypeString.endsWith('*')) {
+    const baseTypeName: TypeName = getSolTypeName(cairoTypeString.slice(0, -1), ast, struct_names);
+    return createArrayTypeName(baseTypeName, ast);
+  }
+  if (cairoTypeString === 'felt') {
+    return createUintNTypeName(248, ast);
+  }
+  if (cairoTypeString === 'Uint256') {
+    return createUint256TypeName(ast);
+  }
+  if (struct_names.includes(cairoTypeString)) {
+    // TODO: replace this with a proper struct type
+    // after creation of struct dependency ordering
+    return new CornerStructTypeName(ast.reserveId(), '', `<STRUCT:${cairoTypeString}>`);
```

Comment:
`getSolTypeName` should be called recursively on the structs and create their replacement type representation in Solidity along with their bounds checking functions.